src/mame/taito/taito_f3_v.cpp

```diff
@@ -1518,110 +1536,55 @@ void taito_f3_state::visible_tile_check(
 }
 
 /******************************************************************************/
+/* window intersection */
+#define CALC_CLIP(p)                                  \
+	clipl = std::max(sa_line->clip##p##_l[y], clipl); \
+	clipr = std::min(sa_line->clip##p##_r[y], clipr);
+
+/* window gap union */
+#define CALC_CLIP_INV(p)                              \
+	clipl = std::min(sa_line->clip##p##_l[y], clipl); \
+	clipr = std::max(sa_line->clip##p##_r[y], clipr);
```

Comment:
Can you turn clipN_l and clipN_r into two-dimensional arrays so these can be turned into lambdas rather than needing token pasting like this?